app/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryManager.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.actors
+
+import akka.actor.{Actor, ActorRef, ActorRefFactory, Props, Scheduler, Terminated}
+import com.typesafe.config.{Config, ConfigFactory}
+import net.ceedubs.ficus.Ficus._
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryConnectorActor.RetryCallToTaxEnrolmentService
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryManager.{MarkTaxEnrolmentRetryRequestAsSuccess, RetryTaxEnrolmentRequest}
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryMongoActor.{DeleteTaxEnrolmentRetryRequest, GetAllNonFailedTaxEnrolmentRetryRequests, UpdateTaxEnrolmentRetryStatusToFail}
+import uk.gov.hmrc.cgtpropertydisposals.models.TaxEnrolmentRequest
+import uk.gov.hmrc.cgtpropertydisposals.repositories.TaxEnrolmentRetryRepository
+import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentService
+import uk.gov.hmrc.cgtpropertydisposals.util.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.duration.{FiniteDuration, _}
+
+class TaxEnrolmentRetryManager(
+  createMongoActor: ActorRefFactory => ActorRef,
+  createConnectorActor: ActorRefFactory => ActorRef,
+  taxEnrolmentRetryRepository: TaxEnrolmentRetryRepository,
+  taxEnrolmentService: TaxEnrolmentService,
+  scheduler: Scheduler
+) extends Actor
+    with Logging {
+  import context.dispatcher
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  val config: Config = ConfigFactory.load()
+
+  val minBackoff: FiniteDuration = config.as[FiniteDuration]("tax-enrolment-retry.min-backoff")
+  val maxBackoff: FiniteDuration = config.as[FiniteDuration]("tax-enrolment-retry.max-backoff")
+  val numberOfRetriesUntilInitialWaitDoubles: Int =
+    config.as[Int]("tax-enrolment-retry.number-of-retries-until-initial-wait-doubles")
+  val maxAllowableElapsedTime: FiniteDuration = config.as[FiniteDuration]("tax-enrolment-retry.max-elapsed-time")
+
+  override def preStart(): Unit = {
+    super.preStart()
+
+    val mongoActorRef     = createMongoActor(context)
+    val connectorActorRef = createConnectorActor(context)
+
+    context.watch(mongoActorRef)
+    context.watch(connectorActorRef)
+
+    mongoActorRef ! GetAllNonFailedTaxEnrolmentRetryRequests
```

Comment:
if this fails to reach the mongo actor then the whole retry mechanism is broken from the start. It'd be safer here to do an ask and get acknowledgement from the mongo actor straightaway before it sends all the retry messages. If the ask times out or the getting from mongo fails then this retry manager can know to retry getting all the failed enrolment requests again